invokeai/app/invocations/generate.py

```diff
@@ -58,29 +58,31 @@ class TextToImageInvocation(BaseInvocation, SDImageInvocation):
 
     # TODO: pass this an emitter method or something? or a session for dispatching?
     def dispatch_progress(
-        self, context: InvocationContext, intermediate_state: PipelineIntermediateState
+        self,
+        context: InvocationContext,
+        source_node_id: str,
```

Comment:
It might be a good idea to just pass `node_id` as the execution graph node. The client receiving this should be able to do the mapping themselves, but would have no way to get the execution graph node this event is for unless you're sending it.